addon/this-fallback-helper.ts

```diff
@@ -0,0 +1,28 @@
+import { helper } from '@ember/component/helper';
+import { deprecate } from '@ember/debug';
+import { get } from '@ember/object';
+
+type Deprecation = Parameters<typeof deprecate>;
+
+type Positional<T, K extends keyof T> = [
+  context: T,
+  path: K,
+  deprecationJson: string | false
+];
+
+/** FIXME */
+const thisFallbackHelper = helper(
+  <T, K extends keyof T>([context, path, deprecationJson]: Positional<
+    T,
+    K
+  >) => {
+    if (deprecationJson) {
+      // FIXME: Assert instead of cast
+      const deprecation = JSON.parse(deprecationJson) as Deprecation;
```

Comment:
This helper is for fallback deprecations that can _only_ be detected at runtime; I needed a way to sneak the `deprecate` call in without futzing up the DOM.

The `deprecationsHelper` helper above is inserted one time, at the bottom of the template, to handle all of the fallback deprecations detected at runtime.

I thought of using this helper in both cases, but since it has potential runtime implications (re: performance at least, maybe bugs) it seemed safer to use it as little as possible.